functions/Set-DbaMaxDop.ps1

```diff
@@ -97,19 +97,25 @@ function Set-DbaMaxDop {
         [switch]$AllDatabases,
         [switch]$EnableException
     )
+    begin {
+        if ($MaxDop -eq -1) {
+            $UseRecommended = $true
+        }
+    }
 
     process {
         if ((Test-Bound -Parameter Database) -and (Test-Bound -Parameter AllDatabases) -and (Test-Bound -Parameter ExcludeDatabase)) {
-            Stop-Function -Category InvalidArgument -Message "-Database, -AllDatabases and -ExcludeDatabase are mutually exclusive. Please choose only one. Quitting."
+            Stop-Function -Category InvalidArgument -Message "-Database, -AllDatabases and -ExcludeDatabase are mutually exclusive. Please choose only one."
             return
         }
 
-        $dbscopedconfiguration = $false
-
-        if ($MaxDop -eq -1) {
-            $UseRecommended = $true
+        if ((Test-Bound -not -Parameter SqlInstance) -and (Test-Bound -not -Parameter InputObject)) {
```

Comment:
```suggestion
        if ((Test-Bound -ParameterName SqlInstance, InputObject -Not)) {
```

Reply:
Didn't know you could do that, very cool.